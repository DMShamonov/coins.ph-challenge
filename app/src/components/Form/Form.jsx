// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { FormPropsType } from './types.js.flow';

export default function Form({
  className,
  children,
  onSubmit,
}: FormPropsType): React.Node {
  return (
    <form className={classnames('form', className)} onSubmit={onSubmit}>
      {children}
    </form>
  );
}
