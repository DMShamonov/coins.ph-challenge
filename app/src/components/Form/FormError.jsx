// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { FormErrorPropsType } from './types.js.flow';

export default function FormError({
  className,
  children,
}: FormErrorPropsType): React.Node {
  return (
    <p className={classnames('form__error', className)}>{children}</p>
  );
}
