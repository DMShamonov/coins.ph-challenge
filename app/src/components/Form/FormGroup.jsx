// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { FormGroupPropsType } from './types.js.flow';

export default function FormGroup({
  className,
  children,
}: FormGroupPropsType): React.Node {
  return (
    <div className={classnames('form__group', className)}>{children}</div>
  );
}
