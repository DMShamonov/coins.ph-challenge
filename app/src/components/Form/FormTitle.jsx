// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { FormTitlePropsType } from './types.js.flow';

export default function FormTitle({
  className,
  children,
}: FormTitlePropsType): React.Node {
  return (
    <p className={classnames('form__title', className)}>{children}</p>
  );
}
