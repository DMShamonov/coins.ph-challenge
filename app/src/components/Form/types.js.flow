// @flow

import type { Node } from 'react';

export type FormPropsType = {|
  className?: string,
  children: Node,
  onSubmit?: Function,
|};

export type FormTitlePropsType = {|
  className?: string,
  children: Node,
|};

export type FormGroupPropsType = {|
  className?: string,
  children: Node,
|};

export type FormErrorPropsType = {|
  className?: string,
  children: Node,
|};
