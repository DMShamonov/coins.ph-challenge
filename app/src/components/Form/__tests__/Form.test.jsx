import React from 'react';
import { shallow } from 'enzyme';

import Form from '../Form';

describe('`Form` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Form>__CONTENT__</Form>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<Form className="__CLASSNAME__">__CONTENT__</Form>)).toMatchSnapshot();
  });
});

describe('Check `Form` props', () => {
  describe('Check callback props', () => {
    test('`onSubmit` called', () => {
      const onSubmit = jest.fn();
      const wrapper = shallow(<Form onSubmit={onSubmit}>__CONTENT__</Form>);

      wrapper.simulate('submit');

      expect(onSubmit).toHaveBeenCalled();
    });
  });
});
