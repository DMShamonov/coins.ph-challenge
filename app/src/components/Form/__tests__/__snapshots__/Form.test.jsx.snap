// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Form\` component should render correctly with \`className\` prop 1`] = `
<form
  className="form __CLASSNAME__"
>
  __CONTENT__
</form>
`;

exports[`\`Form\` component should render correctly with default props 1`] = `
<form
  className="form"
>
  __CONTENT__
</form>
`;
